src/backend/columnar/columnar_reader.c

```diff
@@ -216,6 +217,14 @@ ColumnarReadNextRow(ColumnarReadState *readState, Datum *columnValues, bool *col
 			continue;
 		}
 
+		if (rowNumber)
+		{
+			StripeMetadata *stripeMetadata = list_nth(readState->stripeList,
+													  readState->currentStripe);
```

Comment:
Since we call `ColumnarReadNextRow` during table scan, I wonder if calling `list_nth` for each and every row is bad for performance (before pg13, where lists are linked lists)

Reply:
no, it's not outdated